Create ui-c8y-1023-78-0-add-export-button-visibility-and-permissions-check-for-read-only-users.md (#4650)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/ui-c8y-1023-78-0-add-export-button-visibility-and-permissions-check-for-read-only-users.md

---
date: ""

> title: Export actions disabled for users without inventory write permissions

Reviewer: ```suggestion
title: Export operations disabled for users without appropriate permissions
```

product_area: Application enablement & solutions
change_type:
  - value: change-QHu1GdukP
    label: Feature
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-66664
version: 1023.78.0
---

> Previously, users without inventory write permissions could access all export actions on the Cockpit Exports page and only received an error after attempting to save. This is now fixed. The "Add export" button is disabled with a tooltip explaining insufficient permissions. Duplicate and Delete row actions are hidden for read-only users. The Edit action is replaced with a View action so read-only users can still inspect export details without being able to modify them. In the export detail view, Save buttons are disabled for users without the required permissions.

Reviewer: ```suggestion
Previously, users without Inventory WRITE permission could access all export operations on the **Exports** page and only received an error after attempting to save. This issue has been fixed. 

- For read-only users, the **Add export** button is now disabled, with a tooltip explaining the insufficient permissions. 
- **Duplicate** and **Delete** row commands are hidden. 
-  The **Edit** command is replaced with a **View** command so that read-only users can still inspect export details but cannot modify them. 
- In the export details view, the **Save** buttons are disabled for users without the required permissions.

This change ensures that only users with the appropriate permissions can access export operations, providing a clearer and more secure user experience.
```

---
date: ""

> title: add export button visibility and permissions check for read-only users (#11840) [GRAFT][release/cd] (#11902)

Reviewer 2: ```suggestion
title: Export actions disabled for users without inventory write permissions
```

product_area: Application enablement & solutions
change_type:
  - value: change-QHu1GdukP
    label: Feature
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-66664
version: 1023.78.0
---
# Backport

This will backport the following commits from `develop` to `release/cd`:
- [feat(Web SDK): [MTM-66664] add export button visibility and
permissions check for read-only users
(#11840)](https://github.com/Cumulocity-IoT/cumulocity-ui/pull/11840)

<!--- Backport version: 9.5.1 -->

### Questions ?
Please refer to the [Backport tool
documentation](https://github.com/sorenlouv/backport)

[MTM-66664]:
https://cumulocity.atlassian.net/browse/MTM-66664?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Co-authored-by: Lukasz Janusz <114157358+luja-c8y@users.noreply.github.com>
Co-authored-by: Lukasz Janusz <lukasz.janusz@cumuloicty.com>

> Co-authored-by: Paweł Rynarzewski <92171763+pawel-rynarzewski-c8y@users.noreply.github.com>

Reviewer 2: ```suggestion
Previously, users without inventory write permissions could access all export actions on the Cockpit Exports page and only received an error after attempting to save. This is now fixed. The "Add export" button is disabled with a tooltip explaining insufficient permissions. Duplicate and Delete row actions are hidden for read-only users. The Edit action is replaced with a View action so read-only users can still inspect export details without being able to modify them. In the export detail view, Save buttons are disabled for users without the required permissions.
```
